prepare for v1.4.0

piksel opened the pull request:
this branch contains changes needed for the upcoming release, mainly related to .NET 6

The new target frameworks will be:
- .NET Standard 2.0 (for .NET framework 4.6.2+)
- .NET Standard 2.1 (general support for all but .NET framework)
- .NET 6.0 (new LTS)

the main issue stems from the fact that the AES auth code reading is handled inside a class inheriting from `CryptoStream`. For that to work correctly, the reads cannot be bypassed by more efficient code (which is what is happening in the .NET 6 version of `CryptoStream`).
In fact, the entire `ZIPAESStream` should probably just be removed altogether, since it seems to just be a work around for having the auth code appended to the input stream. this can be handled much easier in the TransformLastBlock method of ZipAESTransform.

that approach is now used by async reads on .NET 6.0, but others still use the legacy code. this was done to do as little as possible to allow SharpZipLib to work with .NET 6, but it really ought to be replaced for all usages.

tonyliving commented:
Can I get it from nuget?

piksel commented:
@tonyliving not until it has been released, no. If you want to try it out you can download the build artifact for this PR:
https://github.com/icsharpcode/SharpZipLib/suites/6639150865/artifacts/250580285

carlreid commented:
No worries, take it easy 😊

carlreid commented:
It seems the artifact from the previous build has expired, is it possible you could re-trigger the build / release this version?

Was just about to host the pre-release package so we could make use of the fix https://github.com/icsharpcode/SharpZipLib/pull/750

piksel commented:
@carlreid I fixed the conflict and there should be a new artifact to download. Also, I fixed the last PR that was blocking release, so it *should* be ready by the weekend.

carlreid commented:
Wonderful! Thank you 😊

In which case, I'll wait until next week to see if you managed to make it through with a release before trying to host the artifact.
